student-test/assignment1/Test_validate_username.c

```diff
@@ -18,5 +18,14 @@ void test_validate_my_username()
      * TODO: Replace the line below with your code here as described above to verify your /conf/username.txt 
      * config file and my_username() functions are setup properly
      */
-    TEST_ASSERT_TRUE_MESSAGE(false,"AESD students, please fix me!");
+
+    //Obtain the Github user-name from the autotest-validate.c file in a string
+    const char * str1 = my_username();
+
+    //Obtain the Github user-name from the conf_file in another string
+    char * str2 = malloc_username_from_conf_file();
```

Comment:
@KhyatiSatta what happens to `str2` after exiting this function?

Reply:
@dwalkes Thanks for the comments!
 Does it have to do with scope and the fact that there is no memory allocated for the string?